base/Const.lua

```diff
@@ -0,0 +1,27 @@
+local META_TABLE    = {
```

Comment:
I suggest `makeReadonly` to create new local metatable with functions unique to each call, referencing local variables from the call. Having a global reference table may be necessary in some places, but we don't need to do it literally every time now.